### ISSUE TITLE
[fastify] Regression #1533: done() is not a function

### ISSUE THREAD

**Issue by Skn0tt** [DataDog/dd-trace-js#1549]
[v1.1.1](https://github.com/DataDog/dd-trace-js/releases/tag/v1.1.1) lead to a crash in my application. It uses fastify@3.20.1 and fastify-swagger@4.5.0. This is the stacktrace:

```
TypeError: done is not a function
at Object.arguments.<computed> (/app/node_modules/dd-trace/packages/datadog-plugin-fastify/src/fastify.js:44:29)
at Object.<anonymous> (/app/node_modules/fastify-swagger/lib/util/common.js:30:7)
at /app/node_modules/fastify/lib/hooks.js:142:22
at Boot.timeoutCall (/app/node_modules/avvio/boot.js:447:5)
at _encapsulateThreeParam (/app/node_modules/avvio/boot.js:545:7)
...
```

This crashed because fastify-swagger expects to receive a `done` function when registering a hook:

https://github.com/fastify/fastify-swagger/blob/4da0ce9def46b971ff2cbe346bbb40d4b9244c6b/lib/util/common.js#L23

I did some digging. This is `fastify.js:44:29`, the second line in the stacktrace:

https://github.com/DataDog/dd-trace-js/blob/f68f74c58c9e506a1eb77afcaaca1a5a1279dabc/packages/datadog-plugin-fastify/src/fastify.js#L44

While `fastify-swagger` expects only a single `done` arg, this line will apply the full `arguments` array instead, no matter what the plugin likes to receive.

This crash is most likely not unique to `fastify-swagger`, but to a lot of fastify plugins. A possible fix would be to take a look at the amount of arguments that the plugin takes, and adapt the applied arguments to it.

I wanted to push a PR with a fix and a reproduction case, but couldn't get the test suite to run properly. So here's an issue :)

**Comment by Skn0tt** [comment]
cc @rochdev

**Comment by rochdev** [comment]
You can isolate the tests for Fastify with either `PLUGINS=fastify yarn test:plugins:ci` or in watch mode with `yarn tdd fastify`. In order to test with the actual `fastify-swagger` library, it can be added to `externals.json` so that it's installed automatically when running these commands.

I will look into it to fix the issue and release a new version as soon as possible (hopefully today), unless you have already started working on it.

**Comment by rochdev** [comment]
Just released the fix in 1.1.2
